colossalai/inference/config.py

```diff
@@ -46,6 +46,7 @@ class InferenceConfig:
         prefill_ratio (Optional[float]): A controling ratio for prefill and decoding in running list, defaults to 1.2. We will do a step of prefill
             when the actual value exceeds this ratio.
         pad_input: Whether to pad all inputs to the max length.
+        return_token_ids: Whether to return output token ids.
```

Comment:
`return_token_ids` is not a really a configuration. It means that whether we want to have this control flow to return extra outputs to the user, so it is more associated with the code logic than generation configurations. I would recommend to do this instead:
```python
class Engine:

    def generate(self, ..., return_token_ids):
          ...

          if return_token_ids:
                  return outputs, token_ids
          else:
                  return outputs
```